pulsar-broker/src/main/java/org/apache/pulsar/broker/web/WebService.java

```diff
@@ -294,8 +291,10 @@ public void start() throws PulsarServerException {
             ContextHandlerCollection contexts = new ContextHandlerCollection();
             contexts.setHandlers(handlers.toArray(new Handler[handlers.size()]));
 
+            Handler handlerForContexts = GzipHandlerUtil.wrapWithGzipHandler(contexts,
```

Comment:
I agree that it is confusing. Refactorings are a different story.

Reply:
I mean, if you're already there, and it's 10 lines, might be worth to change that as well? or you want it the same for cherry-picking?

Reply:
I don't think that such refactorings should be included in this PR. There's a risk of regressions when changing configuration in this area and it would divert the focus of this PR. Please create a separate PR with the changes you have in mind.